vsphere/host_network_system_helper.go

```diff
@@ -45,6 +45,7 @@ func hostVSwitchFromName(client *govmomi.Client, ns *object.HostNetworkSystem, n
 
 	for _, sw := range mns.NetworkInfo.Vswitch {
 		if sw.Name == name {
+			sw.Spec.Mtu = sw.Mtu
```

Comment:
Thats strange because the `mtu` is being read from the `sw.Mtu` instead of `sw.Spec.Mtu`. When testing in my environment, the MTU seems to be properly read without this. Does it happen after every import? If you set the MTU in vCenter, do you still get `0` in terraform?

Reply:
It seems to be related to if the vSwitch is created through vCenter or ESXi. I think the ideal way to fix this will be to change the `read` behavior so that the MTU is pulled from the switch managed object instead of the spec, but that is probably better left for another PR. This seems like a good way to deal with it until then. Thanks for documenting your method so well! It made it really easy to pinpoint the difference.

Reply:
ok